server/src/main/java/org/opensearch/gateway/remote/model/RemoteObject.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote.model;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * An interface to read/write and object from/to a remote storage. This interface is agnostic of the remote storage type.
+ *
+ * @param <T> The object type which can be upload to or download from remote storage.
+ */
+public interface RemoteObject<T> {
```

Comment:
nit: call it `RemoteEntity`?

Reply:
Do we want to tie the interface for cluster state only? The specifications are generic enough to be used for any entity. Can we keep it `RemoteEntity` ?

Reply:
Renamed to RemoteWritableEntity